easy_login/tests/tests.py

```diff
@@ -0,0 +1,110 @@
+            password='user1',
+            email='user1@adm.com'
+        )
+        self.user_2 = User.objects.create_user(
+            username='User_2',
+            password='user2',
+            email='user2@adm.com'
+        )
+        self.user_3 = User.objects.create_user(
+            username='User_3 long name test',
+            password='user3',
+            email='user3@adm.com'
+        )
+
+    def test_user_list_view(self):
+        request = self.factory.get('/')
+        request.user = self.admin
+        response = easy_login(request)
+        with open('template.html', 'w') as ff:
+            ff.write(response['easy_login'])
+        soup = BeautifulSoup(response['easy_login'], features="html.parser")
+        user_name_obj = soup.find('select', {'id': 'id_user_name'})
+        user_real_list = set(User.objects.all().values_list('username', flat=True))
+        user_new_list = {user_obj.text for user_obj in user_name_obj.find_all('option') if user_obj['value']}
+        self.assertEqual(user_new_list, user_real_list)
+
+    def test_current_user(self):
+        request = self.factory.get('/')
+        for user in [self.admin, self.user_1, self.user_2, self.user_3]:
+            request.user = user
+            response = easy_login(request)
+            soup = BeautifulSoup(response['easy_login'], features="html.parser")
+            search_compile = re.compile('Current User: (.*) - ID:.*')
+            user_name = soup.find('p', text=search_compile)
+            user_name = re.findall(search_compile, user_name.text)[0] if user_name else None
+            self.assertEqual(request.user.username, user_name)
+
+    def test_form(self):
+        client = Client()
+        client.login(username=self.admin_username, password=self.admin_password)
+
+        response = client.get('/test_app/')
+        soup = BeautifulSoup(response.content, features="html.parser")
+        search_compile = re.compile('Current User: (.*) - ID:.*')
+        user_name = soup.find('p', text=search_compile)
+        user_name = re.findall(search_compile, user_name.text)[0] if user_name else None
+
+        self.assertEqual('Admin', user_name)
+
+        client.post(
+            '/easy_login/easy_login_change/',
+            {
+                'user_name': '',
+                'user_id': self.user_1.id
+            }
+        )
+
+        response = client.get('/test_app/')
+        soup = BeautifulSoup(response.content, features="html.parser")
+        search_compile = re.compile('Current User: (.*) - ID:.*')
+        user_name = soup.find('p', text=search_compile)
+        user_name = re.findall(search_compile, user_name.text)[0] if user_name else None
+
+        self.assertEqual(self.user_1_username, user_name)
+
+        client.post(
+            '/easy_login/easy_login_change/',
+            {
+                'user_name': self.admin.id,
+                'user_id': ''
+            }
+        )
+
+        response = client.get('/test_app/')
```

Comment:
these tests are too big.
please try split it and try to check something one because it's easier to understand what do you want to test .
and use regexp in the test as for me it's too complex